### ISSUE TITLE
[HOLD for payment 2023-04-03] Never ending loading spinner when tapping on expired magic link

### ISSUE THREAD

**Issue by cristipaval** [Expensify/App#15331]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. The user initiates sign in on the browser
2. The user taps on the magic code link from his inbox
3. A new tab is opened to automatically authenticate the user on the browser
4. The authentication fails for some reason (eg. magic link is expired)

## Expected Result:
The user should be guided to get a new magic code

## Actual Result:
The tab which should authenticate the user shows a never ending loading spinner.

## Workaround:
n/a

## Platforms:
Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0148189905902b2347</li>
        <li>Upwork Job ID: 1628128325668773888</li>
        <li>Last Price Increase: 2023-02-21</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @kevinksullivan (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0148189905902b2347

**Comment by MelvinBot** [comment]
Triggered auto assignment to @shawnborton (`Design`), see [these Stack Overflow questions](https://stackoverflow.com/c/expensify/search?q=%5Bchore%5D%5Bteam-design%5D+is%3Aquestion&searchOn=1) for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @Santhosh-Sellavel (`Internal`)

**Comment by cristipaval** [comment]
@shawnborton I added `Design` label because we need a design for this use case.

**Comment by shawnborton** [comment]
Can you provide a screenshot or screencast of what the current behavior looks like?

**Comment by shawnborton** [comment]
I also wonder if we want to put the user back into the sign in screen when this happens? Something like this?
<img width="360" alt="image" src="https://user-images.githubusercontent.com/2319350/220503633-cf124bcd-db6d-4b2f-abe6-5186f4d3a769.png">

**Comment by cristipaval** [comment]
> Can you provide a screenshot or screencast of what the current behavior looks like?

@shawnborton, Please have a look at the below screen recording were I am switching between tabs.

https://user-images.githubusercontent.com/18078393/220886085-cc47e5d8-1988-4f8e-a3c7-5986890c5f46.mov

**Comment by shawnborton** [comment]
Got it, thanks for sharing that! What do you think about my idea above?

**Comment by cristipaval** [comment]
Hmm.. So if I understand it correctly, it would be as following:
1. The user opens a tab in his browser.
2. Taps on an expired link
3. We redirect the newly opened tab to the sign in form indicating the error
4. The user taps on `Didn't receive a magic code?`
5. Taps on the newly received link
6. The user now is successfully authenticated in the browser, which means that he has 1 tab with the abracadabra message, and 2 tabs with the App authenticated: 1 tab where the entire process started, and another one where we redirected him with the expired link.

Are we fine with the above?

**Comment by shawnborton** [comment]
Hmm maybe I am not following what you are asking then, I didn't realize this was about managing multiple tabs. I thought this was for when a user clicked on a single expired code from their Inbox or something. 

Before diving into design details too much, can you maybe restate the problem and share your recommendation for how to solve it?

**Comment by cristipaval** [comment]
Oh, sorry, I added better [**Action Performed**](https://github.com/Expensify/App/issues/15331#issue-1594052967).

> share your recommendation for how to solve it

My first idea is to adapt the below modal to a failure state and instruct the user to go back to the original device and resend the code. Under the hood, we can also check if the user initiated the sign in on the same browser and also add in the screen a link to resend the magic code.

<img width="620" alt="Screenshot 2023-02-23 at 16 04 32" src="https://user-images.githubusercontent.com/18078393/220930537-cde0e9dc-6ceb-4d42-8090-e6614ec656d1.png">

**Comment by shawnborton** [comment]
Also, one thing I noticed from your screenshot above is that the logo might be getting cut off - can you look into that?
<img width="195" alt="image" src="https://user-images.githubusercontent.com/2319350/220949774-65627c2d-9e1a-49c8-b608-9ba130eae7af.png">

**Comment by cristipaval** [comment]
> Also, one thing I noticed from your screenshot above is that the logo might be getting cut off - can you look into that? <img alt="image" width="195" src="https://user-images.githubusercontent.com/2319350/220949774-65627c2d-9e1a-49c8-b608-9ba130eae7af.png">

hahah, designer's eye 👀 
Yes it looks like it is cut. I'll check it in the code.
Thank you!

**Comment by cristipaval** [comment]
On it today.

**Comment by cristipaval** [comment]
@shawnborton Do we want to show a feedback to the user after clicking the `request a new code here` link?

**Comment by shawnborton** [comment]
Maybe we could just add a new paragraph below that says "Code requested. Please check your device." or something like that once it is clicked?

Here is the svg you need too: [empty-state__toddbehindcloud.svg.zip](https://github.com/Expensify/App/files/10841483/empty-state__toddbehindcloud.svg.zip)

**Comment by cristipaval** [comment]
> Also, one thing I noticed from your screenshot above is that the logo might be getting cut off - can you look into that? <img alt="image" width="195" src="https://user-images.githubusercontent.com/2319350/220949774-65627c2d-9e1a-49c8-b608-9ba130eae7af.png">

@shawnborton It looks like this is how the svg file is in the project assets folder. Could you export it again?

**Comment by shawnborton** [comment]
Sure - can you do me a favor and attach a copy of the current svg here? You will need to .zip it I think. Or just point to it in the code.

**Comment by cristipaval** [comment]
@shawnborton 
Something is weird. [Here](url) you can see it in the code, and looks like github renders it correctly.
Have a look at the screenshot below how it is rendered in vscode.
<img width="1275" alt="Screenshot 2023-03-01 at 16 10 14" src="https://user-images.githubusercontent.com/18078393/222164862-3417a3d5-179b-419d-b989-658db5d69831.png">

**Comment by shawnborton** [comment]
Try this new one: [expensify-wordmark.svg.zip](https://github.com/Expensify/App/files/10861906/expensify-wordmark.svg.zip)

I increased the height of the svg from 18 to 19, not sure if that will require any code changes. But basically the shape was actually sitting outside of the svg bound box once I opened up the file. So hopefully this fixes it!

**Comment by cristipaval** [comment]
Yes! This one works. Thank you!

**Comment by shawnborton** [comment]
Woo, np!

**Comment by MelvinBot** [comment]
@shawnborton, @cristipaval, @kevinksullivan, @Santhosh-Sellavel Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by kevinksullivan** [comment]
PR is WIP

**Comment by cristipaval** [comment]
@shawnborton As we discussed [here](https://expensify.slack.com/archives/C048ZLJ78P5/p1678722446598199?thread_ts=1678573194.799749&cid=C048ZLJ78P5), we need a modal for when the magic link succeeds, but the user isn't fully signed in because 2fa is enabled on his account. In this case, Abracadabra page doesn't make sense.

<img width="1487" alt="Screenshot 2023-03-15 at 00 15 58" src="https://user-images.githubusercontent.com/18078393/225154974-90fc277a-1fd3-416d-b90f-9cc0d5fae5e4.png">

**Comment by shawnborton** [comment]
How about this?
![image](https://user-images.githubusercontent.com/2319350/225406505-11c83043-67cb-43eb-a03e-09a435c5a544.png)


Here is the illustration: [empty-state__safe.svg.zip](https://github.com/Expensify/App/files/10983419/empty-state__safe.svg.zip)

**Comment by Santhosh-Sellavel** [comment]
PR merged

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Santhosh-Sellavel / @cristipaval] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Santhosh-Sellavel / @cristipaval] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Santhosh-Sellavel / @cristipaval] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@kevinksullivan] Determine if we should create a regression test for this bug.
- [ ] [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@kevinksullivan] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by NikkiWines** [comment]
@cristipaval @Santhosh-Sellavel @kevinksullivan just bumping this issue - anything further that we need to do here to get this closed out?

**Comment by cristipaval** [comment]
> The PR that introduced the bug has been identified
This is a follow-up issue for automatic authentication with a magic link. We implemented the happy scenario initially and this issue was for the corner cases.

**Comment by cristipaval** [comment]
> @cristipaval @Santhosh-Sellavel @kevinksullivan just bumping this issue - anything further that we need to do here to get this closed out?

I think the payment is the only thing left.

**Comment by MelvinBot** [comment]
⚠️ Looks like this issue was linked to a **Deploy Blocker** [here](https://github.com/Expensify/App/issues/17091)

If you are the assigned CME please investigate whether the linked PR caused a regression and leave a comment with the results.

If a regression has occurred and you are the assigned CM follow the instructions [here](https://stackoverflow.com/c/expensify/questions/13647).

If this regression could have been avoided please consider also proposing a recommendation to the PR checklist so that we can avoid it in the future.

**Comment by kevinksullivan** [comment]
Regardless of this ^, we would've paid this out. The job closed so I just sent you a new offer @Santhosh-Sellavel . Let me know when you've accepted.

**Comment by shawnborton** [comment]
Not overdue

**Comment by kevinksullivan** [comment]
Paid @Santhosh-Sellavel
